Build the latest go-jsonnet with -mod=vendor failed

I'm trying to upgrade go-jsonnet in nixpkgs. But I got a failed buld.
### Steps to Reproduce
```bash
$ git clone https://github.com/google/go-jsonnet
$ cd go-jsonnet
$ go build -mod=vendor ./cmd/...
# golang.org/x/sys/unix
vendor/golang.org/x/sys/unix/syscall.go:83:16: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
vendor/golang.org/x/sys/unix/syscall_linux.go:2255:9: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
vendor/golang.org/x/sys/unix/syscall_unix.go:118:7: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
vendor/golang.org/x/sys/unix/sysvshm_unix.go:33:7: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
```

--- a/go.mod
+++ b/go.mod
@@ -1,11 +1,16 @@
 module github.com/google/go-jsonnet
 
-go 1.13
+go 1.17
 
 require (
 	github.com/fatih/color v1.12.0
 	github.com/sergi/go-diff v1.1.0
+	sigs.k8s.io/yaml v1.1.0
+)
+
+require (
+	github.com/mattn/go-colorable v0.1.8 // indirect
+	github.com/mattn/go-isatty v0.0.12 // indirect
 	golang.org/x/sys v0.1.0 // indirect
 	gopkg.in/yaml.v2 v2.2.7 // indirect
-	sigs.k8s.io/yaml v1.1.0
 )

